grpc/errors.go

```diff
@@ -63,59 +72,83 @@ func unwrapError(err error, md metadata.MD) error {
 		return nil
 	}
 
-	unwrappedErr := status.Convert(err).Message()
-
 	errTypeStrs, ok := md["errortype"]
 	if !ok {
 		return err
 	}
+
+	inErrMsg := status.Convert(err).Message()
 	if len(errTypeStrs) != 1 {
 		return berrors.InternalServerError(
-			"multiple errorType metadata, wrapped error %q",
-			unwrappedErr,
+			0,
+			"multiple 'errortype' metadata, wrapped error %q",
+			inErrMsg,
 		)
 	}
 
-	errType, decErr := strconv.Atoi(errTypeStrs[0])
+	inErrType, decErr := strconv.Atoi(errTypeStrs[0])
 	if decErr != nil {
 		return berrors.InternalServerError(
+			0,
 			"failed to decode error type, decoding error %q, wrapped error %q",
 			decErr,
-			unwrappedErr,
+			inErrMsg,
 		)
 	}
-	outErr := berrors.New(berrors.ErrorType(errType), unwrappedErr)
-
-	subErrsJSON, ok := md["suberrors"]
-	if !ok {
-		return outErr
-	}
-	if len(subErrsJSON) != 1 {
-		return berrors.InternalServerError(
-			"multiple suberrors metadata, wrapped error %q",
-			unwrappedErr,
+	inErr := berrors.New(berrors.ErrorType(inErrType), 0, inErrMsg)
+	var outErr *berrors.BoulderError
+	if !errors.As(inErr, &outErr) {
+		return fmt.Errorf(
+			"expected type of inErr to be %T got %T: %q",
+			outErr,
+			inErr,
+			inErr.Error(),
 		)
 	}
 
-	var suberrs []berrors.SubBoulderError
-	err2 := json.Unmarshal([]byte(subErrsJSON[0]), &suberrs)
-	if err2 != nil {
-		return berrors.InternalServerError(
-			"error unmarshaling suberrs JSON %q, wrapped error %q",
-			subErrsJSON[0],
-			unwrappedErr,
-		)
+	subErrorsVal, ok := md["suberrors"]
+	if ok {
+		if len(subErrorsVal) != 1 {
+			return berrors.InternalServerError(
+				0,
+				"multiple 'suberrors' in metadata, wrapped error %q",
+				inErrMsg,
+			)
+		}
+
+		var subErrs []berrors.SubBoulderError
+		unmarshalErr := json.Unmarshal([]byte(subErrorsVal[0]), &subErrs)
```

Comment:
Can this be simplified to
```suggestion
		unmarshalErr := json.Unmarshal([]byte(subErrorsVal[0]), &outErr.SubErrors)
```
(and then also delete line 130)? This would make this stanza and the retryAfter stanza below more similar.